docs/kb/function-tutorial-deploy.md

```diff
@@ -0,0 +1,212 @@
+
+<img src="./img/kn-ps-slack-events.png" width="80%" align="center" class="border m-1 p-1"/>
+
+
+## Deploy kn-ps-slack Function to the VEBA Appliance
+1.  Move to the /vcenter-event-broker-appliance/examples/knative/powershell/kn-ps-slack directory that you cloned earlier with git.  
+
+2.  Update the slack_secret.json file with your Slack webhook URL.
+
+3.  Create the Kubernetes secret which can then be accessed from within the function by using the environment variable named SLACK_SECRET.
+
+```
+# create secret
+kubectl -n vmware-functions create secret generic slack-secret --from-file=SLACK_SECRET=slack_secret.json
+
+# update label for secret to show up in VEBA UI
+kubectl -n vmware-functions label secret slack-secret app=veba-ui
+```
+
+4.  Edit the function.yaml file with the name of the custom container image you pushed to DockerHub. For the example listed above, we would use:
+
+```
+spec:
+  containers:
+    - image: docker.io/atauber/kn-ps-slack:1.1
+```
+
+By default, the function deployment will filter on the VmPoweredOffEvent vCenter server event. If you wish to change this event type, update the subject field within function.yaml to the desired event type.  The function.yaml file is shown below.  vCenter server events are described here: [vCenter Events](https://vmweventbroker.io/kb/vcenter-events).
+
+<img src="./img/kn-ps-slack-function.png" width="80%" align="center" class="border m-1 p-1"/>
+
+
+5.  Deploy the function to the VEBA Appliance.
+
+```
+# deploy function
+kubectl -n vmware-functions apply -f function.yaml
+```
+
+Now with the new function deployed, we should see these resources in the vmware-functions namespace:
+
+<img src="./img/kn-ps-slack-function3.png" width="80%" align="center" class="border m-1 p-1"/>
+
+
+If we then create a VM named "prod-testvm" and power on / power off, we should have success!
+
+<img src="./img/kn-ps-slack-alert2.png" width="60%" align="center" class="border m-1 p-1"/>
+
+
+You will also be able to see the alert in Sockeye by searching for the VmPoweredOffEvent:
+
+<img src="./img/kn-ps-slack-alert3.png" width="100%" align="center" class="border m-1 p-1"/>
+
+
+## Kubernetes Troubleshooting commands
+When things don't work as expected, it is usefull to know how to troubleshoot a system.  Help in troubleshooting the VEBA appliance and functions can be found here:
+- [Troubleshoot Appliance](troubleshoot-appliance)
+- [Troubleshoot Functions](troubleshoot-functions)
+
+We have used kubectl get and apply in the above examples.  The following are some of the more common kubectl commands used in troubleshooting.  For the describe and exec commands, we will use an ubuntu pod as our test case. Use the following command to deploy the ubuntu pod:
```

Comment:
why not use your function example? using ubuntu is confusing IMHO bc it uses a different namespace, apply is new to the user and it really is then not about debugging functions.

Reply:
Fixed by using the veba-ui pod instead.